vortex-serde/src/file/reader/filtering.rs

```diff
@@ -3,7 +3,7 @@ use vortex_error::VortexResult;
 
 use super::projections::Projection;
 
-pub trait FilteringPredicate {
+pub trait FilteringPredicate: Send + Sync {
```

Comment:
Datafusion basically requires it

Reply:
Should be `Send + 'static`, I'll fix that in a separate PR as it requires some unwinding around the IO code

Reply:
This will change in my pr. The api here doesn't work for our serialization format.